### PR TITLE
Add TaskViewModel and use cases

### PR DESCRIPTION
**Pull request by AndreVero**

### CODE REVIEW

File: app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsEvent.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.vero.tasky.agenda.presentation.taskdetails
+
+import com.vero.tasky.agenda.presentation.model.ReminderRange
+import java.time.LocalDate
+import java.time.LocalTime
+
+sealed interface TaskDetailsEvent {
+    object ChangeMode: TaskDetailsEvent
+    class CheckModifiedInfo(val title: String?,
```
Comment (MKiperszmid) [inline app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsEvent.kt:9]: What's the use if this event?

File: app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsEvent.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+                            val description: String?): TaskDetailsEvent
+    class AtDateChanged(val date: LocalDate) : TaskDetailsEvent
+    class AtTimeChanged(val time: LocalTime) : TaskDetailsEvent
+    class ReminderChanged(val reminderRange: ReminderRange) : TaskDetailsEvent
+    object SaveEvent : TaskDetailsEvent
```
Comment (MKiperszmid) [inline app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsEvent.kt:14]: SaveTask*

File: app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsEvent.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    object DeleteEvent : TaskDetailsEvent
```
Comment (MKiperszmid) [inline app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsEvent.kt:15]: DeleteTask*

File: app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsScreen.kt [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+@Composable
+fun TaskDetailsScreen(
+    viewModel: TaskDetailsViewModel = hiltViewModel(),
+    onEditTitle: (String) -> Unit,
+    onEditDescription: (String) -> Unit,
+    navigateBack: () -> Unit,
+    title: String? = null,
+    description: String? = null,
+    deletedPhotoUri: String? = null,
```
Comment (MKiperszmid) [inline app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsScreen.kt:15]: Tasks dont have photos

File: app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+import javax.inject.Inject
+
+@HiltViewModel
+class TaskDetailsViewModel @Inject constructor(
+    private val taskUseCases: TaskUseCases,
+    private val savedStateHandle: SavedStateHandle
+) : ViewModel() {
+
+    private val itemId = savedStateHandle.get<String?>(NavigationConstants.ITEM_ID)
+        ?: savedStateHandle.get(NavigationConstants.TASK_ITEM_ID)
```
Comment (MKiperszmid) [inline app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsViewModel.kt:31]: Why would you sometimes have ITEM_ID and others TASK_ITEM_ID?

File: app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsEvent.kt (hunk 1 above)
Comment (AndreVero) [reply app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsEvent.kt:9]: The same as in EventScreen, if the user changes the title or description I need to pass them from navBackEntry to the screen, and then from LauncedEffect to viewModel (and for this I'm using this event)

File: app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsViewModel.kt (hunk 5 above)
Comment (AndreVero) [reply app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsViewModel.kt:31]: It's some kind of legacy, currently, I can change this. (ITEM_ID was used for itemId that was passed through the agenda screen and TASK_ITEM_ID was used for navDeepLink)

File: app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsEvent.kt (hunk 1 above)
Comment (MKiperszmid) [reply app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsEvent.kt:9]: Ahh gotcha

File: app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsViewModel.kt (hunk 5 above)
Comment (MKiperszmid) [reply app/src/main/java/com/vero/tasky/agenda/presentation/taskdetails/TaskDetailsViewModel.kt:31]: Ah
I would unify it into just one

**Pull request merged**